### PR TITLE
[DDP] fix test_ddp_inference

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61666**

Closes https://github.com/pytorch/pytorch/issues/61481. Fixes this
test by removing section that uses only torch.no_grad() and doesn't call
model.eval(). For SyncBN, need to call model.eval() otherwise SyncBN will
assume it is in training mode, which does collective calls in the forward pass
and does not work for inference.

Differential Revision: [D29699444](https://our.internmc.facebook.com/intern/diff/D29699444/)

### CODE REVIEW

File: torch/testing/_internal/distributed/distributed_test.py [hunk 1]
```diff
@@ -7112,14 +7111,8 @@ def test_ddp_inference(self):
             for test in tests:
                 test_model, test_local_model, test_inp = test
                 if self.rank == 0:
-                    with torch.no_grad():
```
Comment (mrshenli) [inline torch/testing/_internal/distributed/distributed_test.py:7115]: Does this mean `no_grad` is no longer relevant for inference and `.eval()` will sufficient for all cases?

**Review by mrshenli — APPROVED** [review]

**Comment by rohan-varma** [comment]
Failure is unrelated - `FAIL [21.893s]: test_fd_sharing (__main__.TestMultiprocessing)`
